src/mon/Monitor.cc

```diff
@@ -3329,40 +3329,63 @@ void Monitor::handle_route(MonOpRequestRef op)
   }
 }
 
+namespace {
+  struct ReentrantLock {
+    const bool initially_locked;
+    Mutex &mutex;
+    ReentrantLock(Mutex& m)
+      : initially_locked(m.is_locked_by_me()),
```

Comment:
I don't think we should ever use is_locked* as a conditional--only for debug assertions.  Notably, this is not available from std::mutex.